### ISSUE TITLE
[BUG] TypeError: Cannot read property 'text' of undefined

### ISSUE THREAD

**Issue by LeithOscar** [compodoc/compodoc#1068]
Hi!
executing this command to generate doc =>  compodoc -p tsconfig.json -s

**TypeScript version used by Compodoc : 2.9.1
Node.js version : v12.14.1
Operating system : Windows 10**

`[15:50:56] Analysing routes definitions and clean them if necessary
Unhandled Rejection at: Promise {
  <rejected> TypeError: Cannot read property 'text' of undefined
      at C:\Users\xxxxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:5691:55
      at Function.ArrayUtils.find (C:\Users\xxxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\node_modules\ts-simple-ast\dist\utils\ArrayUtils.js:38:21)
      at Object.getNodeByNameOrFindFunction (C:\Users\xxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\node_modules\ts-simple-ast\dist\utils\compiler\getNodeByNameOrFindFunction.js:10:36)
      at SourceFile.StatementedNode.class_1.getVariableDeclaration (C:\Users\xxxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\node_modules\ts-simple-ast\dist\compiler\statement\StatementedNode.js:312:28)
      at RouterParserUtil.cleanCallExpressions (C:\Users\xxxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:5688:45)
      at AngularDependencies.getSourceFileDecorators (C:\Users\xxxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:8151:46)
      at C:\Users\xxxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:7960:31
      at Array.map (<anonymous>)
      at AngularDependencies.getDependencies (C:\Users\xxxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:7949:21)
      at CliApplication.Application.getDependenciesData (C:\Users\xxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:9498:40)
} reason: TypeError: Cannot read property 'text' of undefined
    at C:\Users\xxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:5691:55
    at Function.ArrayUtils.find (C:\Users\xxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\node_modules\ts-simple-ast\dist\utils\ArrayUtils.js:38:21)
    at Object.getNodeByNameOrFindFunction (C:\Users\xxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\node_modules\ts-simple-ast\dist\utils\compiler\getNodeByNameOrFindFunction.js:10:36)
    at SourceFile.StatementedNode.class_1.getVariableDeclaration (C:\Users\xxxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\node_modules\ts-simple-ast\dist\compiler\statement\StatementedNode.js:312:28)
    at RouterParserUtil.cleanCallExpressions (C:\Users\xxxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:5688:45)
    at AngularDependencies.getSourceFileDecorators (C:\Users\xxxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:8151:46)
    at C:\Users\xxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:7960:31
    at Array.map (<anonymous>)
    at AngularDependencies.getDependencies (C:\Users\xxxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:7949:21)
    at CliApplication.Application.getDependenciesData (C:\Users\xxx\AppData\Roaming\npm\node_modules\@compodoc\compodoc\dist\application-54cd2170.js:9498:40)
[15:50:56] Sorry, but there was a problem during parsing or generation of the documentation. Please fill an issue on github. (https://github.com/compodoc/compodoc/issues/new)`

**Comment by ManuelRauber** [comment]
Check, if you have template literal types somewhere in your project. They have been introduced in TypeScript 4.1 (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#template-literal-types).

In our project that leads to the exact same error.

**Comment by vogloblinsky** [comment]
@ManuelRauber Do you have an example file where it breaks ? Thanks

**Comment by ManuelRauber** [comment]
@vogloblinsky Yes!

We have that little piece of code: 

```
type DigitWithoutZero = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9;
type DigitWithZero = 0 | DigitWithoutZero;

export type Foo =
  | `${DigitWithoutZero}`
  | `${DigitWithoutZero}${DigitWithZero}`;
```

That will break compodoc.

**Comment by vogloblinsky** [comment]
@ManuelRauber Ok it is fixed in last development version.
@LeithOscar Could you give me the source code of the file where it breaks ?

**Comment by ManuelRauber** [comment]
@vogloblinsky Thanks! Would it be possible to get development versions on npm? like `1.2.0-dev.XXXX` whereas X is a build number or something.

**Comment by parkersweb** [comment]
Just to add - we have a very similar issue:

```
export type Foo = 
  | 'apple'
  | 'pear'
  | 'orange';

export type Bar = Foo | null;

export type FooBar = [Bar, ...Bar[]];
```
Causes compodoc to fall over with this error:

```
Unhandled Rejection at: Promise {
  <rejected> TypeError: Cannot read property 'text' of undefined
      at ClassHelper.visitTypeName (/Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:6467:22)
      at ClassHelper.visitTypeName (/Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:6470:21)
      at ClassHelper.visitType (/Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:6556:45)
      at parseNode (/Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:8395:56)
      at /Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:8575:13
      at visitNodes (/Users/xxx/node_modules/ts-simple-ast/node_modules/typescript/lib/typescript.js:14786:30)
      at Object.forEachChild (/Users/xxx/node_modules/ts-simple-ast/node_modules/typescript/lib/typescript.js:15012:24)
      at AngularDependencies.getSourceFileDecorators (/Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:8155:16)
      at /Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:7960:31
      at Array.map (<anonymous>)
} reason: TypeError: Cannot read property 'text' of undefined
    at ClassHelper.visitTypeName (/Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:6467:22)
    at ClassHelper.visitTypeName (/Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:6470:21)
    at ClassHelper.visitType (/Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:6556:45)
    at parseNode (/Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:8395:56)
    at /Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:8575:13
    at visitNodes (/Users/xxx/node_modules/ts-simple-ast/node_modules/typescript/lib/typescript.js:14786:30)
    at Object.forEachChild (/Users/xxx/node_modules/ts-simple-ast/node_modules/typescript/lib/typescript.js:15012:24)
    at AngularDependencies.getSourceFileDecorators (/Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:8155:16)
    at /Users/xxx/node_modules/@compodoc/compodoc/dist/application-54cd2170.js:7960:31
    at Array.map (<anonymous>)
```

**Comment by vogloblinsky** [comment]
@parkersweb Fixed for your code in 1.1.14

**Comment by parkersweb** [comment]
Thanks so much @vogloblinsky !
